python-spec/src/somacore/collection.py

```diff
@@ -20,34 +34,47 @@ class Collection(base.SOMAObject, MutableMapping[str, _ST], metaclass=abc.ABCMet
 
     __slots__ = ()
 
+    @classmethod
     @abc.abstractmethod
-    def add(
+    def create(
+        cls: Type[_CT],
+        uri: str,
+        *,
+        platform_config: Optional[options.PlatformConfig] = None,
+        context: Optional[Any] = None,
+    ) -> _CT:
+        """Creates a new Collection at the given URI and returns it.
+
+        The collection will be returned in the opened state.
+        """
+        raise NotImplementedError()
+
+    @abc.abstractmethod
+    def add_collection(
```

Comment:
the names `create()` vs `add_foo()` seem slightly in opposition.  In both cases, a new persistent object is created, but the term `add` doesn't really convey that (to me).

Assuming our term for construction is `create`, I only have bad naming suggestions, but here goes:
* create_child_collection()
* create_add_collection()

I don't think we should hold up the PR for this - unless somebody has a good naming idea. But `add` didn't really say "i am a factory" method to me.

Reply:
@bkmartinjr @thetorpedodog `add_collection` -> `make_collection` maybe -- ?

Reply:
I was definitely feeling this as I wrote this but did not have a better idea at the time. Coming back with fresh eyes, I think `add_new_whatever` is a pretty good way to go.